ext/date/php_date.c

```diff
@@ -5296,7 +5296,14 @@ static zval *date_period_read_property(zval *object, zval *member, int type, voi
 /* {{{ date_period_write_property */
 static void date_period_write_property(zval *object, zval *member, zval *value, void **cache_slot)
 {
-	zend_throw_error(NULL, "Writing to DatePeriod properties is unsupported");
+	if (!std_object_handlers.has_property(object, member, 2, NULL)) {
```

Comment:
Doesn't that mean that `$datePeriod->notExistingProperty = 123` is also forbidden? Not that I'm a fan of dynamic properties, but it is odd to enforce that here. I also think this is going to interact badly with typed properties in their uninitialized state (or unset normal declared properties).

I think this should just explicitly check against all the "special" properties instead.

Reply:
Good spot, I've hard-coded the list of properties here now and just preventing writing to those.
I've also updated the bug report test to allow writing dynamic properties